scripts/generate-types.js

```diff
@@ -54,9 +56,16 @@ export namespace Webhooks {
 
     public on (event: 'error', callback: (event: Error) => void): void
     public on (event: '*' | string[], callback: (event: Webhooks.WebhookEvent<any>) => void): void
+    public on (event: '*' | string[], callback: (event: Webhooks.WebhookEvent<any>) => Promise<void>): void
     ${signatures.join('\n')}
 
     public sign (data: any): string
+    public verify (eventPayload: WebhookPayloadWithRepository, signature: string): boolean
```

Comment:
Sorry, I’m not sure what `WebhookPayloadWithRepository` is? Not all event payload has a repository key, e.g. https://developer.github.com/v3/activity/events/types/#installationevent has `repositories`.

But if that works, can we use the same type for `sign()`? Both `sign()` and `verify()` accept the same payload type as first argument

Reply:
I had taken these from a PR of mine from Probot. I'll fix that. I didn't notice it